src/instructlab/train/linux_train.py

```diff
@@ -289,31 +288,30 @@ def model_generate(user, **kwargs):
     )
 
     tokenizer.padding_side = "right"
-    output_dir = "./training_results"
-    per_device_train_batch_size = 1
+    train_args = Munch(
+        output_dir="./training_results",
+        fp16=four_bit_quant,
+        num_train_epochs=num_epochs,
+        per_device_train_batch_size=1,
+        save_strategy="epoch",
+        report_to="none",
+    )
     max_seq_length = 300
 
     if device.type == "hpu":
         # Intel Gaudi trainer
         # https://docs.habana.ai/en/latest/PyTorch/Getting_Started_with_PyTorch_and_Gaudi/Getting_Started_with_PyTorch.html
         # https://huggingface.co/docs/optimum/habana/quickstart
         # https://huggingface.co/docs/optimum/habana/package_reference/gaudi_config
-        if per_device_train_batch_size == 1:
-            per_device_train_batch_size = 8
-
-        training_arguments = GaudiTrainingArguments(
-            output_dir=output_dir,
-            num_train_epochs=num_epochs,
-            per_device_train_batch_size=per_device_train_batch_size,
-            bf16=True,
-            save_strategy="epoch",
-            report_to="none",
-            use_habana=True,
-            use_lazy_mode=True,
-            # create checkpoint directories
-            save_on_each_node=True,
-            # gaudi_config_name=gaudi_config_name,
-        )
+        gta = GaudiTrainingArguments(**train_args, use_habana=True)
+        gta.bf16 = True
+        gta.use_lazy_mode = True
+        # create checkpoint directories
+        gta.save_on_each_node = True
+        # gaudi_config_name=gaudi_config_name,
+        if gta.per_device_train_batch_size == 1:
+            gta.per_device_train_batch_size = 8
```

Comment:
Have you verified that this approach works with Intel Gaudi software 1.15.1 and 1.16.0, and Intel Gaudi 2 cluster?

Reply:
No.

I've just assured than result of `GaudiTrainingArguments` is the same as `before`:

```
import dictdiffer

for diff in list(dictdiffer.diff(gta.to_dict(), training_arguments.to_dict()):
    print(diff)
```